File: backend/qms-system/src/main/java/nz/ac/autuni/qmssystem/model/FMEATable.java
package nz.ac.autuni.qmssystem.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import javax.persistence.GeneratedValue;
import java.io.Serializable;

import static javax.persistence.GenerationType.IDENTITY;

/**
 * @author Simon-the-coder
 * @date 3/09/20 5:34 pm
 */
@AllArgsConstructor
@NoArgsConstructor
@Data
@Document(collection = "FMEATable")
public class FMEATable implements Serializable {
    @Id @GeneratedValue(strategy = IDENTITY)
    private String hazardId;
    private String hazardClass;
    private String sourceId;
    private String foreseeableSequenceOfEvents; //both normal and fault conditions
    private String hazardousSituation;
    private String harm; //result of hazardous situation
    private String severityOfHarm; //1-7
    private String probability; // Probability of hazardous situation occurring, Probability of hazardous situation leading to harm (1-7)
    private String riskPriority; //Risk Priority Number (SxP)
    private String recommendingAction; //Risk Control Method
    private String typeOfAction;
    private String actionDone;
    private String severityOfHarm2;
    private String probability2;
    private String residualRisk; //(S2xP2)
    private String selectedRPId;
    private boolean acceptability;
}

File: frontend/QMS_system/lib/model/fmea_table.dart
class FMEATable {
  String selectedRPId; // selected risk procedure Id
  String hazardId;
  String hazardClass;
  String sourceId;
  String foreseeableSequenceOfEvents; //both normal and fault conditions
  String hazardousSituation;
  String harm; //result of hazardous situation
  String severityOfHarm; //1-7
  String probability; // Probability of hazardous situation occurring, Probability of hazardous situation leading to harm (1-7)
  String riskPriority; //Risk Priority Number (SxP)
  String recommendingAction; //Risk Control Method
  String typeOfAction;
  String actionDone;
  String severityOfHarm2;
  String probability2;
  String residualRisk; //(S2xP2)
  bool acceptability;

  FMEATable({
    this.selectedRPId,
    this.hazardId,
    this.hazardClass,
    this.sourceId,
    this.foreseeableSequenceOfEvents,
    this.hazardousSituation,
    this.harm,
    this.severityOfHarm,
    this.probability,
    this.riskPriority,
    this.recommendingAction,
    this.typeOfAction,
    this.actionDone,
    this.severityOfHarm2,
    this.probability2,
    this.residualRisk,
    this.acceptability,
  });

  FMEATable.fromJson(Map<String, dynamic> json)
    : selectedRPId = json['selectedRPId'],
      hazardId = json['hazardId'],
      hazardClass = json['hazardClass'],
      sourceId = json['sourceId'],
      foreseeableSequenceOfEvents = json['foreseeableSequenceOfEvents'],
      hazardousSituation = json['hazardousSituation'],
      harm = json['harm'],
      severityOfHarm = json['severityOfHarm'],
      probability = json['probability'],
      riskPriority = json['riskPriority'],
      recommendingAction = json['recommendingAction'],
      typeOfAction = json['typeOfAction'],
      actionDone = json['actionDone'],
      severityOfHarm2 = json['severityOfHarm2'],
      probability2 = json['probability2'],
      residualRisk = json['residualRisk'],
      acceptability = json['acceptability'];

  Map<String, dynamic> toJson() => {
    'selectedRPId' : selectedRPId,
    'hazardId': hazardId,
    'hazardClass': hazardClass,
    'sourceId': sourceId,
    'foreseeableSequenceOfEvents': foreseeableSequenceOfEvents,
    'hazardousSituation': hazardousSituation,
    'harm': harm,
    'severityOfHarm': severityOfHarm,
    'probability': probability,
    'riskPriority': riskPriority,
    'recommendingAction': recommendingAction,
    'typeOfAction': typeOfAction,
    'actionDone': actionDone,
    'severityOfHarm2': severityOfHarm2,
    'probability2': probability2,
    'residualRisk': residualRisk,
    'acceptability': acceptability,
  };
}
